gcsfs/core.py

```diff
@@ -152,7 +152,7 @@ def validate_response(r, path):
         except:
             msg = str(r.content)
 
-        if "Not Found" in m:
+        if "Not Found" in m or 'No such object' in m:
             raise FileNotFoundError(path)
```

Comment:
Is there any way to make these checks less fragile?  For example is there a status code that is both sufficiently meaningful and less likely for google to change?